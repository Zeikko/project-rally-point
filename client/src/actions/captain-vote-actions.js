import * as actions from '../../../common/actions.json'

export const voteCaptainAction = (gameId, playerId, userId) => ({
  type: actions.CAPTAIN_VOTE_REQUEST,
  socket: true,
  playerId,
  gameId,
  userId,
})
