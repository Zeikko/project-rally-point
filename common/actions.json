{
  "GET_USER_REQUEST": "GET_USER_REQUEST",
  "GET_USER_SUCCESS": "GET_USER_SUCCESS",
  "GET_USER_ERROR": "GET_USER_ERROR",

  "LOGOUT_REQUEST": "LOGOUT_REQUEST",
  "LOGOUT_SUCCESS":"LOGOUT_SUCCESS",
  "LOGOUT_ERROR": "LOGOUT_ERROR",

  "JOIN_GAME_REQUEST": "JOIN_GAME_REQUEST",
  "JOIN_GAME_SUCCESS": "JOIN_GAME_SUCCESS",
  "JOIN_GAME_ERROR": "JOIN_GAME_ERROR",

  "LEAVE_GAME_REQUEST": "LEAVE_GAME_REQUEST",
  "LEAVE_GAME_SUCCESS": "LEAVE_GAME_SUCCESS",
  "LEAVE_GAME_ERROR": "LEAVE_GAME_ERROR",

  "GET_GAME_REQUEST": "GET_GAME_REQUEST",
  "GET_GAME_SUCCESS": "GET_GAME_SUCCESS",
  "GET_GAME_ERROR": "GET_GAME_ERROR",

  "GET_PLAYERS_REQUEST": "GET_PLAYERS_REQUEST",
  "GET_PLAYERS_SUCCESS": "GET_PLAYERS_SUCCESS",
  "GET_PLAYERS_ERROR": "GET_PLAYERS_ERROR",

  "CAPTAIN_VOTE_REQUEST": "CAPTAIN_VOTE_REQUEST",
  "CAPTAIN_VOTE_SUCCESS": "CAPTAIN_VOTE_SUCCESS",
  "CAPTAIN_VOTE_ERROR": "CAPTAIN_VOTE_ERROR",

  "GET_CAPTAIN_VOTES_REQUEST": "GET_CAPTAIN_VOTES_REQUEST",
  "GET_CAPTAIN_VOTES_SUCCESS": "GET_CAPTAIN_VOTES_SUCCESS",
  "GET_CAPTAIN_VOTES_ERROR": "GET_CAPTAIN_VOTES_ERROR",

  "CREATE_GAME_REQUEST": "CREATE_GAME_REQUEST",
  "CREATE_GAME_SUCCESS": "CREATE_GAME_SUCCESS",
  "CREATE_GAME_ERROR": "CREATE_GAME_ERROR",

  "PICK_PLAYER_REQUEST": "PICK_PLAYER_REQUEST",
  "PICK_PLAYER_SUCCESS": "PICK_PLAYER_SUCCESS",
  "PICK_PLAYER_ERROR": "PICK_PLAYER_ERROR",

  "SIMULATE_USER": "SIMULATE_USER"

}
