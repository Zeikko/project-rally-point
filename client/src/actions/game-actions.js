import * as actions from '../../../common/actions.json'

export const joinGameAction = (id) => ({
  type: actions.JOIN_GAME_REQUEST,
  socket: true,
  id
})

export const leaveGameAction = (id) => ({
  type: actions.LEAVE_GAME_REQUEST,
  socket: true,
  id
})
