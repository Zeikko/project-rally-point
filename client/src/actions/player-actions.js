import * as actions from '../../../common/actions.json'

export const joinGameAction = (gameId, userId) => ({
  type: actions.JOIN_GAME_REQUEST,
  socket: true,
  gameId,
  userId,
})

export const leaveGameAction = (gameId, userId) => ({
  type: actions.LEAVE_GAME_REQUEST,
  socket: true,
  gameId,
  userId,
})
