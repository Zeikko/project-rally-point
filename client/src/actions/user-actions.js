import actions from '../../../common/actions.json'

export const getUserAction = () => ({
  type: actions.GET_USER_REQUEST,
})

export const logoutAction = () => ({
  type: actions.LOGOUT_REQUEST,
})
