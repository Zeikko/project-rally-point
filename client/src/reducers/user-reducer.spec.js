import userReducer from './user-reducer'
import actions from '../../../common/actions.json'
import initialState from '../initial-state'

describe('userReducer', () => {
  it('returns old state', () => {
    const nextState = userReducer(undefined, {
      type: 'irrelevant action',
    })
    expect(nextState).toEqual(initialState.userState)
  })

  it('simulates user', () => {
    const nextState = userReducer(undefined, {
      type: actions.SIMULATE_USER,
      id: 1,
    })
    expect(nextState).toMatchSnapshot()
  })
})
