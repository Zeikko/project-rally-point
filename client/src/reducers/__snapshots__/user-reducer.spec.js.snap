// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer simulates user 1`] = `
Object {
  "isLoading": false,
  "user": Object {
    "displayName": "Test user 1",
    "id": 1,
    "profileUrl": "http://steamcommunity.com/id/1",
    "role": "simulated user",
    "smallAvatarUrl": "https://www.gravatar.com/avatar/1?s=32&d=identicon&r=PG",
    "steamId": "1",
  },
}
`;
